### PR TITLE
RSC Can Add Another RSC to manage assignments

### PR DESCRIPTION
**Pull request by angiecstn**
### PT Link
https://www.pivotaltracker.com/story/show/161620321

### Included in PR

- Added devise gem
- generated login views with devise
- configure User model
- added user controller with a create action that allows current user to add a new user
- Feature test using FactoryBot with 3 scenarios(including sad paths)
- Adding view for login/signup page 
- Adding links to dashboard leading to login

### Screenshots

screenshot 2018-11-01 at 15 47 35
screenshot 2018-11-01 at 15 47 04
screenshot 2018-11-01 at 15 48 47

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+         :recoverable, :rememberable, :validatable
+end
+
+def self.from_devise(params)
+    where(email: params.email).first_or_create do |user|
+    user.password = "password"
+    user.password_confirmation = "password"
+    binding.pry
+    User.create!(user)
+  end
```
Comment (RhonGabriel) [inline app/models/user.rb]: remove this action as it's unnecessary at this stage. We'll work on another feature with the functionality of creating a random password to be sent to the new user

**Review by RhonGabriel — APPROVED** [review]

File: app/views/devise/registrations/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<h2>Edit <%= resource_name.to_s.humanize %></h2>
```
Comment (rupertlion) [inline app/views/devise/registrations/edit.html.erb]: I know we don't use this here, but... if we manually remove it, I understand we can only generate views in devise based on module scope (eg all views in registrations) which might then override our styled new view. Probably a question for Thomas.

File: app/views/users/create.html.haml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+%h1 Users#create
```
Comment (rupertlion) [inline app/views/users/create.html.haml]: Perhaps remove this content as not really relevant

File: app/views/users/index.html.haml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+%h1 You are on the user index
```
Comment (rupertlion) [inline app/views/users/index.html.haml]: Perhaps retitle to be relevant

File: config/routes.rb [hunk 5]
```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
-  root controller: :jobs, action: :index 
-  resources :hospitals, only: :index
+  root controller: :users, action: :index 
+    resources :users, only: [:create, :new]
+  devise_for :users
+  
+    resources :jobs, only: :index
```
Comment (rupertlion) [inline config/routes.rb]: should these resources be indented? I think maybe indent should align with the scope line below

Comment (RhonGabriel) [reply config/routes.rb]: solved

File: features/rsc_can_register_to_account.feature [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    Given the following registered RSC exist
+            | email           | password   | 
+            | email@email.com | mypassword |
+    And the RSC user is logged in as "email@email.com"
+    And I am on the landing page
+    And I click on "Create RSC account"
+    Then I am on the create RSC page
+
+    Scenario: RSC can sucessfully add another RSC user to access the platform
+        Then stop
```
Comment (Noelryn) [inline features/rsc_can_register_to_account.feature]: Could you remove the then stop????

**Review by tochman — APPROVED** [review]
Okay. But you have messed up the git flow (see the other branch) You CAN NOT pull from each other to different branches.

**Pull request merged**